File: calculus/euler_approximation/c/euler.c
#include <stdio.h> 
#define LOW_BOUND 0
#define HIGH_BOUND 3
#define h 0.125 
double f(double x, double y);
int main () 
{ 	int a = (int)((HIGH_BOUND - LOW_BOUND) / h) + 1;
	double x[a], y[a]; 
	int i;
	x[0] = 0.0;
	y[0] = 1.0; 
	for (i = 0; i < a; i++){
		x[i + 1] = x[i] + h;
		y[i + 1] = y[i] + h * f(x[i], y[i]);
	} 
	printf("x[i]\ty[i]\n"); 
	for (i = 0; i < a; i++)
		printf("%.4f\t%.4f\n", x[i], y[i]); 
	return 0;
} 
double f(double x, double y) 
{ 
	return(3 * (x + y));
} 

File: calculus/euler_approximation/python/euler.py
print('Enter lower bound')
LOW_BOUND = int(input())
print('Enter higher bound')
HIGH_BOUND = int(input())
h = 0.125

def func(x,y):
	return 3*(x+y);

a = int((HIGH_BOUND-LOW_BOUND)//h) + 1
x = [0.0]
y = [1.0]
for i in range(0,a):
	x.append(x[i] + h)
	y.append(y[i] + h*func(x[i],y[i]))

print(" x[i]\t y[i]\n")
for i in range(0,a):
	print("{0:.4f}\t{0:.4f}\n".format(x[i],y[i]))

